asserts/snap_asserts.go

```diff
@@ -678,11 +702,36 @@ func assembleSnapRevision(assert assertionBase) (Assertion, error) {
 		return nil, err
 	}
 
+	integrityMap, err := checkMap(assert.headers, "integrity")
+	if err != nil {
+		return nil, err
+	}
+
+	var integrity *Integrity
+
+	if integrityMap != nil {
+		_, err := checkDigest(integrityMap, "sha3-384", crypto.SHA3_384)
```

Comment:
given the nesting we need to introduce What variant of this helper, see checkIntWhat, checkOptionalStringWhat for what is meant by that

Reply:
ha I knew there was info missing in my errors there, but completely missed the What variants. Thanks, added.